doc/grpc.md

```diff
@@ -0,0 +1,109 @@
+# GRPC on APICast
+
+On Apicast 3.8 full HTTP2 protocol is supported, this enable APICast to handle
+GRPC connections end to end taking care APICast about Authentication, headers
+modification, etc..
+
+HTTP2 is only enabled in the TLS port, and it’s not enabled in the plaintext
+port. The main reason why is only enabled in TLS is to make sure that both HTTP
+1.1 and HTTP2 protocols are supported. ALPN(Application-Layer Protocol
+Negotiation) decides within the TLS hello exchanges what protocol should be
+used. HTTP2 always takes precedence.
+
+For authentication, headers authentication or openID (Json Web tokens)  is
+highly recommended, so it’ll be more transparent for the GRPC endpoints and it’s
+easy to add in the GRPC clients.
+
+The GRPC endpoint server should also listen in TLS, MTLS can be enabled or not,
+but the endpoint should finish TLS connections to be able to work as expected.
+
+The only work needed in APICast to enable full GRPC traffic is to enable the
+GRPC policy in the given service, so the protocol used will be full HTTP2. This
+is because nginx directive does not support ALPN at all.
```

Comment:
**Suggestion:** In APIcast, for full gRPC traffic, enable the gRPC policy in the given service. The protocol used will then be full HTTP/2. This is because  NGINX directives do not support ALPN.